packages/front-end/components/Features/CodeSnippetModal.tsx

```diff
@@ -137,559 +111,274 @@ export default function CodeSnippetModal({
-
-// Use a feature!
-if (growthbook.isOn(${JSON.stringify(featureId)})) {
-  // ...
-}
-`.trim()}
-          />
-        </Tab>
-        <Tab display="React" id="tsx">
-          <p>
-            Read the <DocLink docSection="tsx">full React docs</DocLink> for
-            more details.
-          </p>
-          <Code
-            language="sh"
-            code="npm i --save @growthbook/growthbook-react"
-          />
-          <Code
-            language="tsx"
-            code={`
-import { GrowthBook, GrowthBookProvider, useFeature } from "@growthbook/growthbook-react";
-import { useEffect } from "react";
+    // connection changes & current language isn't included in new connection, reset to default
+    if (!currentConnection.languages.includes(language)) {
+      setLanguage(currentConnection.languages[0] || "javascript");
+    }
+  }, [currentConnection]);
 
-// Create a GrowthBook instance
-const growthbook = new GrowthBook({
-  // enableDevMode: true allows you to use the Chrome DevTools Extension to test/debug.
-  enableDevMode: true,
-  trackingCallback: (experiment, result) => {
-    ${indentLines(
-      getTrackingCallback(
-        state.tracking,
-        state.gaDimension + "",
-        "experiment.key",
-        "result.variationId"
-      ),
-      4
-    )}
+  if (!currentConnection) {
+    return null;
   }
-});
-
-export default function MyApp() {
-  useEffect(() => {
-    // Load feature definitions from API${
-      isCloud()
-        ? ""
-        : `\n    // In production, we recommend putting a CDN in front of the API endpoint`
-    }
-    fetch("${getSDKEndpoint(apiKey)}")
-      .then((res) => res.json())
-      .then((json) => {
-        growthbook.setFeatures(json.features);
-      });
 
-    // TODO: replace with real targeting attributes
-    growthbook.setAttributes(${indentLines(stringify(exampleAttributes), 4)})
-  }, [])
+  const { docs, label, usesEntireEndpoint } = languageMapping[language];
+  const apiHost = getApiBaseUrl(currentConnection);
+  const clientKey = currentConnection.key;
+  const featuresEndpoint = apiHost + "/api/features/" + clientKey;
+  const encryptionKey =
+    currentConnection &&
+    currentConnection.encryptPayload &&
+    currentConnection.encryptionKey;
 
   return (
-    <GrowthBookProvider growthbook={growthbook}>
-      <MyComponent/>
-    </GrowthBookProvider>
-  )
-}
+    <>
+      {showTestModal && setShowTestModal && (
+        <CheckSDKConnectionModal
+          close={() => {
+            mutateConnections();
+            setShowTestModal(false);
+          }}
+          connection={currentConnection}
+          mutate={mutateConnections}
+          goToNextStep={submit}
+        />
+      )}
+      <Modal
+        close={close}
+        secondaryCTA={secondaryCTA}
+        bodyClassName="p-0"
+        open={true}
+        inline={inline}
+        size={"lg"}
+        header="Implementation Instructions"
```

Comment:
nit: on the `[sdkid].tsx` implementation, the font for this header is smaller than some of the sub-headers, such as "Javascript Config Settings" or "Installation"